lib/floe/workflow/states/choice.rb

```diff
@@ -21,11 +21,19 @@ def initialize(workflow, name, payload)
         def finish(context)
           input      = input_path.value(context, context.input)
           output     = output_path.value(context, input)
+          # For a bad path, throws an ExecutionError
           next_state = choices.detect { |choice| choice.true?(context, output) }&.next || default
 
+          runtime_field_error!("Default", nil, "not defined and no match found") unless next_state
+
           context.next_state = next_state
           context.output     = output
           super
+        rescue Floe::ExecutionError => e
+          # input / output paths were bad
+          # choice state was bad
+          context.next_state = nil
+          context.output     = {"Error" => "States.RuntimeError", "Cause" => e.message}
```

Comment:
Would this be a good time to use `States.NoChoiceMatched`?
> A Choice State failed to find a match for the condition field extracted from its input.

Reply:
this caused a little more change than I wanted - but great idea.